nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1544,6 +1544,15 @@ def build_transformer_config(self) -> TransformerConfig:
 
         attention_softmax_in_fp32 = False  # not currently used in NeMo unless apply_query_key_layer_scaling is True
         apply_query_key_layer_scaling = self.cfg.get('apply_query_key_layer_scaling', False)
+
+        fp16_enabled = self.trainer.precision in [16, '16', '16-mixed']
+        if apply_query_key_layer_scaling:
+            if fp16_enabled:
+                os.environ["NVTE_APPLY_QK_LAYER_SCALING"] = "1"
+            else:
+                logging.warning("apply_query_key_layer_scaling is only enabled when using FP16, setting it to False")
```

Comment:
Should we set the env var here as well to "0" ?

Reply:
good point, otherwise it will error 366cf61f82ada5e89f5af5c45eed598309ec6875